api/v1alpha1/resourcegraphdefinition_types.go

```diff
@@ -33,6 +35,35 @@ type ResourceGraphDefinitionSpec struct {
 	Resources []*Resource `json:"resources,omitempty"`
 }
 
+type ResourceGraphDefinitionReconcileSpec struct {
+	// InstancePolicy defines the reconciliation policy for the instances
+	// created from the ResourceGraphDefinition.
+	// When set to "Periodic", or not set, the instances will be reconciled periodically.
+	// When set to "Reactive", the instances will be reconciled when any changes are detected
+	// on the resources owned by the instance. This change detection is achieved by watching
+	// the resources owned by the instance and enqueueing a reconcile request for the controller
+	// responsible for the instance.
+	//
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default:=Periodic
+	// +kubebuilder:validation:Enum=Periodic;Reactive
+	InstancePolicy ResourceGraphDefinitionInstancePolicy `json:"instancePolicy,omitempty"`
```

Comment:
I think this shouldn't be exposed.

There are at least 2 cases which makes the current behaviour unpredictable. Either when the KRO pod is restarted or when the RGD is updated. Both cases trigger a reconciliation of each instance and hence override any changes on the children.

Making KRO always reactive helps bring predictability in the system as any manual change on the children would be overridden almost immediately, just as soon as the child is updated.
With this, I would expect the person doing the manual change to wonder why the manual changes were overridden rather than taking the manual change for done.

Reply:
still i agree that default should probably be reactive.

Reply:
The more i think about it, the more i'm leaning towards making "Reactive" the default behaviour, and not expose this configuration at all.

Reply:
Moved to reactive by default, resolving and fixing tests separately